src/Time/TimeSteppers/TimeStepper.hpp

```diff
@@ -0,0 +1,71 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class TimeStepper.
+
+#pragma once
+
+#include <deque>
+#include <tuple>
+
+#include "Time/Time.hpp"
+#include "Utilities/FakeVirtual.hpp"
+
+/// Holds classes that take time steps.
+namespace TimeSteppers {}
+
+namespace TimeSteppers {
+class AdamsBashforthN;
+class RungeKutta3;
+}  // namespace TimeSteppers
+
+namespace TimeStepper_detail {
+DEFINE_FAKE_VIRTUAL(update_u)
+}  // TimeStepper_detail
+
+/// Abstract base class for TimeSteppers.
+class TimeStepper /*: public Factory<TimeStepper>*/ {
+ public:
+  using Inherit = TimeStepper_detail::FakeVirtualInherit_update_u<TimeStepper>;
+  static std::string class_id() noexcept { return "TimeStepper"; }
+  using creatable_classes = typelist<
+      TimeSteppers::AdamsBashforthN,
+      TimeSteppers::RungeKutta3>;
+
+  /// \cond HIDDEN_SYMBOLS
+  TimeStepper() noexcept = default;
+  TimeStepper(const TimeStepper&) noexcept = default;
+  TimeStepper& operator=(const TimeStepper&) noexcept = default;
+  TimeStepper(TimeStepper&&) noexcept = default;             // NOLINT
+  TimeStepper& operator=(TimeStepper&&) noexcept = default;  // NOLINT
+  virtual ~TimeStepper() noexcept = default;
+  /// \endcond
+
+  /// Add the change for the current substep to u and return the substep size.
+  /// More recent values should be later in the history list.
```

Comment:
More recent in wall clock time, or something along those lines.  The statement is trying to say it's being used in a queue-like manner and specifying which end to push things onto.  Unfortunately, `std::queue` is missing most of the container operations and so is unusable here, even though logically it would be clearer.